neuraxle/metaopt/auto_ml.py

```diff
@@ -0,0 +1,358 @@
+        :rtype: HyperparameterSamples
+        """
+        hyperparams, scores = self.load_all_trials()
+
+        best_score = None
+        best_hyperparams = None
+
+        for trial_hyperparam, trial_score in zip(hyperparams, scores):
+            if best_score is None or higher_score_is_better == (trial_score > best_score):
+                best_score = trial_score
+                best_hyperparams = trial_hyperparam
+
+        return best_hyperparams
+
+    def _get_trial_hash(self, hp_dict):
+        current_hyperparameters_hash = hashlib.md5(str.encode(str(hp_dict))).hexdigest()
+        return current_hyperparameters_hash
+
+
+class HyperparamsJSONRepository(HyperparamsRepository):
+    """
+    Hyperparams repository that saves json files for every AutoML trial.
+
+    .. seealso::
+        :class:`HyperparamsRepository`,
+        :class:`HyperparameterSamples`,
+        :class:`AutoMLSequentialWrapper`
+    """
+
+    def __init__(self, folder):
+        self.cache_folder = folder
+
+    def create_new_trial(self, hyperparams: HyperparameterSamples):
+        """
+        Create new hyperperams trial json file.
+
+        :param hyperparams: hyperparams
+        :type hyperparams: HyperparameterSamples
+        :return:
+        """
+        score = None
+        self._save_trial_json(hyperparams, score)
+
+    def load_all_trials(self) -> Tuple[List[HyperparameterSamples], List[float]]:
+        """
+        Load all hyperparameter trials with their corresponding score.
+        Reads all the saved trial json files.
+
+        :return: (hyperparams, scores)
+        """
+        all_hyperparams = []
+        all_scores = []
+
+        for base_path in glob.glob(os.path.join(self.cache_folder, '*.json')):
+            with open(base_path) as f:
+                trial_json = json.load(f)
+
+            all_hyperparams.append(HyperparameterSamples(trial_json['hyperparams']))
+            all_scores.append(trial_json['score'])
+
+        return all_hyperparams, all_scores
+
+    def save_score_for_trial(self, hyperparams: HyperparameterSamples, score: float):
+        """
+        Save hyperparams, and score for a trial.
+
+        :return: (hyperparams, scores)
+        """
+        self._save_trial_json(hyperparams, score)
+
+    def _save_trial_json(self, hyperparams, score):
+        """
+        Save trial json file.
+
+        :return: (hyperparams, scores)
+        """
+        hp_dict = hyperparams.to_flat_as_dict_primitive()
+        current_hyperparameters_hash = self._get_trial_hash(hp_dict)
+        with open(os.path.join(self.cache_folder, current_hyperparameters_hash) + '.json', 'w+') as outfile:
+            json.dump({
+                'hyperparams': hp_dict,
+                'score': score
+            }, outfile)
+
+
+class AutoMLStrategyMixin:
+    """
+    Base class for Automatic Machine Learning strategies.
+    Implement your own custom intelligent search of hyperparameters to get most accurate predictive models.
```

Comment:
We instead need an `AutoMLWrapper` as in this issue: https://github.com/Neuraxio/Neuraxle/issues/45

Note that the wrapper as coded in the issue is generic and can: 
- adapt to any repository
- adapt to any pipeline
- adapt to any AutoML algorithm

This way, it should be possible to use a Random Search inside the `AutoMLWrapper` without creating a `RandomSearchBaseAutoMLStrategy`.

Reply:
I don't know. Maybe you wanted something else by coding this, I may simply just not understand what you did and it might be okay.

Reply:
have you read the code ?

Reply:
I didn't want to do something else. I have tried what was in the issue, and I found my solution better.

Reply:
the random search step is not clean as it is right now. It contains a loop that shouldn't be there. the auto ml strategy for the random search should only be one line : 

hyperparams.rvs()

Reply:
I don't want to reuse the Random Search step because it is not clean.

Reply:
Note that the wrapper as coded in the issue is generic and can:

- adapt to any repository
- adapt to any pipeline
- adapt to any AutoML algorithm

???? AutoMLSequentialWrapper does just that as well

Reply:
the only thing that differs in all auto ml strategy is how to find the next best hyperparams, and how to find the best hyperparams from all the trials

Reply:
please read AutoMLStrategyMixin....